exasol/analytics/query_handler/graph/stage/sql/input_output.py

```diff
@@ -14,12 +15,10 @@ class SQLStageInputOutput:
     For example, a dependency could be a table which the previous stage computed and
     the subsequent one uses.
     """
+    pass
 
-    dataset: Dataset
-    dependencies: Dependencies = dataclasses.field(default_factory=dict)
-    """
-    This contains user-defined dependencies which the previous stage wants to communicate to the subsequent stage.
-    """
 
-    def __post_init__(self):
-        check_dataclass_types(self)
+@dataclass(frozen=True)
+class MultiDatasetSQLStageInputOutput(SQLStageInputOutput):
+    datasets: Dict[object, Dataset]
```

Comment:
Instead of a single dataset (potentially containing multiple partitions) we now have a dict of potentially multiple datasets indexed by arbitrary objects.